string building
string floor
---
miltbot_map/Waypoint[] waypoints